experiments/ssl4eo/download_ssl4eo.py

```diff
@@ -163,78 +161,63 @@ def adjust_coords(
     ]
 
 
-def get_properties(image: ee.Image) -> Any:
-    return image.getInfo()
-
-
 def get_patch(
     collection: ee.ImageCollection,
     center_coord: list[float],
     radius: float,
     bands: list[str],
-    crop: Optional[dict[str, Any]] = None,
+    original_resolutions: list[int],
+    new_resolutions: list[int],
     dtype: str = "float32",
 ) -> dict[str, Any]:
-    image = collection.sort("system:time_start", False).first()  # get most recent
-    region = (
-        ee.Geometry.Point(center_coord).buffer(radius).bounds()
-    )  # sample region bound
-    reproj_bands = "B8"
-    reproj_resolution = 30
-    other_bands = [b for b in bands if b != reproj_bands]
-    patch_reproj = (
-        image.select(reproj_bands)
-        .reproject(
-            crs=image.select(reproj_bands).projection().crs(), scale=reproj_resolution
-        )
-        .sampleRectangle(region, defaultValue=0)
-    )
-    patch_other = image.select(*other_bands).sampleRectangle(region, defaultValue=0)
-    # patch = image.select(*bands).sampleRectangle(region, defaultValue=0)
-    # features = patch.getInfo()  # the actual download
-    features_reproj = patch_reproj.getInfo()
-    features_other = patch_other.getInfo()
-
-    raster = OrderedDict()
-    for band in bands:
-        if band == reproj_bands:
-            img = np.atleast_3d(features_reproj["properties"][band])
-        else:
-            img = np.atleast_3d(features_other["properties"][band])
-        if crop is not None:
-            img = center_crop(img, out_size=crop[band])
-        raster[band] = img.astype(dtype)
-
-    coords0 = np.array(features_other["geometry"]["coordinates"][0])
+    image = collection.sort("CLOUD_COVER").first()
```

Comment:
In my mind, it's better to sort by least cloud cover than by more recent images.

Reply:
Does it support sorting by both like pandas does?

Reply:
No idea, but cloud cover is float so I doubt we'll ever find two that match.